studio/components/interfaces/Settings/Database/SSLConfiguration.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  FormHeader,
+  FormPanel,
+  FormSection,
+  FormSectionContent,
+  FormSectionLabel,
+} from 'components/ui/Forms'
+import { useProjectSettingsQuery } from 'data/config/project-settings-query'
+import { useSSLEnforcementQuery } from 'data/ssl-enforcement/ssl-enforcement-query'
+import { useSSLEnforcementUpdateMutation } from 'data/ssl-enforcement/ssl-enforcement-update-mutation'
+
+const SSLConfiguration = () => {
+  const { ui } = useStore()
+  const { ref } = useParams()
+  const [isEnforced, setIsEnforced] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const { data: projectSettings } = useProjectSettingsQuery({ projectRef: ref })
+  const { data: sslEnforcementConfiguration, isLoading } = useSSLEnforcementQuery({
+    projectRef: ref,
+  })
+  const { mutateAsync: updateSSLEnforcement } = useSSLEnforcementUpdateMutation()
+
+  const hasAccessToSSLEnforcement = !sslEnforcementConfiguration?.isNotAllowed
+  const env = process.env.NEXT_PUBLIC_ENVIRONMENT === 'prod' ? 'prod' : 'staging'
+  const hasSSLCertificate =
+    projectSettings?.project !== undefined &&
+    new Date(projectSettings.project.inserted_at) >= new Date('2021-04-30')
+
+  useEffect(() => {
+    if (!isLoading && sslEnforcementConfiguration) {
+      setIsEnforced(
+        sslEnforcementConfiguration.appliedSuccessfully &&
+          sslEnforcementConfiguration.currentConfig.database
+      )
+    }
+  }, [isLoading])
+
+  const toggleSSLEnforcement = async () => {
+    if (!ref) return console.error('Project ref is required')
+
+    setIsEnforced(!isEnforced)
+    setIsSubmitting(true)
+
+    try {
+      await updateSSLEnforcement({
+        projectRef: ref,
+        requestedConfig: { database: !isEnforced },
+      })
+    } catch (error: any) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to update SSL enforcement: ${error.message}`,
+      })
+      setIsEnforced(isEnforced)
+    } finally {
+      setIsSubmitting(false)
+    }
+  }
+
+  if (!hasAccessToSSLEnforcement) return <></>
+
+  return (
+    <div>
+      <div className="flex items-center justify-between">
+        <FormHeader title="SSL Configuration" description="" />
+        <div className="flex items-center space-x-2 mb-6">
+          <Link href="https://supabase.com/docs/guides/platform/ssl-enforcement">
+            <a target="_blank">
+              <Button type="default" icon={<IconExternalLink />}>
+                Documentation
+              </Button>
+            </a>
+          </Link>
+        </div>
+      </div>
+      <FormPanel>
+        <FormSection
+          header={
+            <FormSectionLabel
+              className="lg:col-span-7"
+              description={
+                <div className="space-y-4">
+                  <p className="text-sm text-scale-1000">
+                    Reject connections made to your database without a valid certificate
```

Comment:
`without a valid certificate` makes it sound like we're enforcing client cert auth, which we're not. Would suggest "Rejects non-SSL connections to your database" or "Rejects unsecure connections to your database" to keep it simple.